core-processor/src/executor.rs

```diff
@@ -182,7 +182,14 @@ where
         AllocationsContext::new(allocations.clone(), static_pages, settings.max_pages);
 
     // Creating message context.
-    let message_context = MessageContext::new(dispatch.clone(), program_id, msg_ctx_settings);
+    let Some(message_context) = MessageContext::new(dispatch.clone(), program_id, msg_ctx_settings)
```

Comment:
whoa, I was missing it. I'll open an issue from this comment to remove cloning here because it could be pretty harmful

Reply:
I don't think it's harmful, but pretty stupid yes